power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -329,16 +340,91 @@ template <bool sym> class YBus {
         auto const& y_bus_entry_indptr = y_bus_struct_->y_bus_entry_indptr;
         // loop for each y bus position
         for (Idx entry = 0; entry != nnz(); ++entry) {
+            // start admittance accumulation with zero
+            ComplexTensor<sym> entry_admittance{0.0};
+            IdxVector entry_param_shunt;
+            IdxVector entry_param_branch;
+            //   loop over all entries of this position
+            for (Idx element = y_bus_entry_indptr[entry]; element != y_bus_entry_indptr[entry + 1]; ++element) {
+                auto param_idx = y_bus_element[element].idx;
+                if (y_bus_element[element].element_type == YBusElementType::shunt) {
+                    // shunt
+                    entry_admittance += shunt_param[param_idx];
+                    entry_param_shunt.push_back(param_idx);
+                } else {
+                    // branch
+                    entry_admittance +=
+                        branch_param[param_idx].value[static_cast<Idx>(y_bus_element[element].element_type)];
+                    entry_param_branch.push_back(param_idx);
+                }
+            }
+            // assign
+            admittance[entry] = entry_admittance;
+            map_admittance_param_branch.push_back(entry_param_branch);
+            map_admittance_param_shunt.push_back(entry_param_shunt);
+        }
+        // move to shared ownership
+        admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
+    }
+
+    IdxVector increments_to_entries(auto const& math_model_param_incrmt) {
+        // construct affected entries
+        IdxVector affected_entries;
+
+        // query params in map, not yet distinguishing between entry_param_shunt_pair and entry_param_shunt
```

Comment:
i don't think you need this comment. the name is sufficiently clear and the usage is straight-forward

```suggestion
```

Reply:
Resolved in latest commit.